test/integration/targets/loops/tasks/main.yml

```diff
@@ -176,3 +176,29 @@
   with_sequence: start=0 count=3
   loop_control:
     index_var: my_idx
+
+#
+# loop_control/label
+# https://github.com/ansible/ansible/pull/36430
+#
+
+- set_fact:
+    loopthis:
+        - name: foo
+          label: foo_label
+        - name: bar
+          label: bar_label
+
+- name: check that item label is updated each iteration
+  debug:
+    msg: "{{ looped_var.name }}"
+  with_items: "{{ loopthis }}"
```

Comment:
How about `with_dict`?

Reply:
`loopthis` is a `list`, I don't see how would changing it to `dict` and using `with_dict` make testing the loop label better?